django/templatetags/i18n.py

```diff
@@ -135,6 +135,8 @@ def render(self, context, nested=False):
         singular, vars = self.render_token_list(self.singular)
         if self.plural and self.countervar and self.counter:
             count = self.counter.resolve(context)
+            if not isinstance(count, int):
+                raise TemplateSyntaxError(f'"count" in \'{self.tag_name}\' tag must be an integer.')
```

Comment:
As far as I'm aware `count` works fine with any numbers not only integers, see [comment](https://code.djangoproject.com/ticket/18995#comment:2):
```suggestion
            if not isinstance(count, (decimal.Decimal, float, int)):
                raise TemplateSyntaxError("'count' argument to %r tag must be a number.' % self.tag_name)
```

Reply:
Thanks, good catch.